core/thread/fetch_spec.rb

```diff
@@ -0,0 +1,30 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+ruby_version_is '2.5' do
+  describe 'Thread#fetch' do
+    it 'returns the value of the fiber-local varible if value has been assigned' do
+      th = Thread.new { Thread.current[:cat] = 'meow' }
+      th.join
+      th.fetch(:cat, true).should == 'meow'
+    end
+
+    it "returns the default value if fiber-local variable hasn't been assigned" do
+      th = Thread.new {}
+      th.join
+      th.fetch(:cat, true).should == true
+    end
+
+    it 'raises a KeyError when only one argument is passed' do
+      th = Thread.new {}
+      th.join
+      -> { th.fetch(:cat) }.should raise_error(KeyError)
+    end
+
+    it 'raises an ArgumentError when not passed one or two arguments' do
+      th = Thread.new {}
+      th.join
+      -> { th.fetch() }.should raise_error(ArgumentError)
+      -> { th.fetch(1, 2, 3) }.should raise_error(ArgumentError)
```

Comment:
Could you use `Thread.current` for these examples?

Reply:
yes, I can use Thread.current. I have updated the PR.